resources/resource/resource_helpers.go

```diff
@@ -55,6 +57,12 @@ func getParam(r Resource, key string, stringToLower bool) any {
 		return cast.ToFloat64(v)
 	case time.Time:
 		return val
+	case toml.LocalDate, toml.LocalDateTime:
+		val, err := htime.ToTimeInDefaultLocationE(val, time.UTC)
```

Comment:
This is nit picking, but but when you have a known type, converting it to another type (e.g. `interface{}` or `reflect.Value`) just so it can be converted back to the type you started with. I don't think it allocates any extra memory in this particular case, but I had to navigate into the ToTime... func to see what it does. I suggest you just call val.AsTime directly.

Reply:
Sorry about the iterations here. This was the self-inflicted wound:
```
case toml.LocalDate, toml.LocalDateTime:
```

<https://go.dev/ref/spec#Type_switches>

> In clauses with a case listing exactly one type, the variable has that type; otherwise, the variable has the type of the expression in the TypeSwitchGuard.

Which meant I had type `any` instead of the toml.Whatever type, which meant AsTime() was not an option. Split the case; all is good.